tests/rest/client/v1/test_login.py

```diff
@@ -144,3 +147,45 @@ def test_POST_ratelimiting_per_account_failed_attempts(self):
         self.render(request)
 
         self.assertEquals(channel.result["code"], b"403", channel.result)
+
+    def test_soft_logout(self):
+        self.register_user("kermit", "monkey")
+
+        # we shouldn't be able to make requests without an access token
+        request, channel = self.make_request(b"GET", TEST_URL)
+        self.render(request)
+        self.assertEquals(channel.result["code"], b"401", channel.result)
+        self.assertEquals(channel.json_body["errcode"], "M_MISSING_TOKEN")
+
+        # log in as normal
+        params = {
+            "type": "m.login.password",
+            "identifier": {"type": "m.id.user", "user": "kermit"},
+            "password": "monkey",
+        }
+        request, channel = self.make_request(b"POST", LOGIN_URL, params)
+        self.render(request)
+
+        self.assertEquals(channel.result["code"], b"200", channel.result)
+
+        # we should now be able to make requests with the access token
+        access_token = channel.json_body["access_token"]
+        request, channel = self.make_request(
+            b"GET", TEST_URL, access_token=access_token
+        )
+        self.render(request)
+        self.assertEquals(channel.result["code"], b"200", channel.result)
+
+        # time passes
+        self.reactor.advance(24 * 3600 * 1000)
+
+        # ... and we should be soft-logouted
```

Comment:
can we also get a test/assert for repeated calls with the logged out token returning `soft_logout: true`? Just to ensure that noisy clients don't see soft_logout -> logout and wipe people's keys.

Reply:
I've added more tests in 545a9e9.